### PR TITLE
Removed google cloud storage headers from response

### PR DESCRIPTION
**Pull request by Syphon83**

### CODE REVIEW

File: templates/fragments/osb.conf [hunk 1]
```diff
@@ -11,6 +11,15 @@ location ~* /(s3fs-css|s3fs-js|${DRUPAL_PUBLIC_FILES_PATH})/(.*) {
...
     proxy_pass $bucket_base_url/$file_path$is_args$args;
+
+    #Hide Headers comin up from the storage bucket. 
+    proxy_hide_header "x-guploader-uploadid";
+    proxy_hide_header "x-goog-generation";
+    proxy_hide_header "x-goog-metageneration";
+    proxy_hide_header "x-goog-metageneration";
+    proxy_hide_header "x-goog-stored-content-length";
+    proxy_hide_header "x-goog-hash";
+    proxy_hide_header "x-goog-storage-class";
```
Comment (paolomainardi) [inline templates/fragments/osb.conf]: generally speaking this kind of stuff should be toggled by a condition, there could be cases where we would like to have those variables (for example in a dev environment).

We should use an approach like this one:

https://github.com/sparkfabrik/docker-php-drupal-nginx/blob/feature/d8/templates/default.conf#L37
```
if [ ${NGINX_HTTPSREDIRECT} == 1 ]; then
  print "Enabling HTTPS redirect"
  sed  -e '/#httpsredirec/r /templates/httpsredirect.conf' -i /templates/default.conf;
  sed  -e '/#httpsredirec/r /templates/httpsredirect.conf' -i /templates/subfolder.conf;
fi
```

Comment (Syphon83) [reply templates/fragments/osb.conf]: Understood. I'm going with it. Thanks

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -63,11 +64,18 @@ fi
 # If we are using an Object Storage Bucket, we add a custom location file.
 # We also check if a file with the same name does not exist, to prevent the override.
 if [ ! -z ${NGINX_OSB_BUCKET} ] && [ ! -f "/etc/nginx/conf.d/fragments/osb.conf" ]; then
+  
+  # If we want to suppress google headers coming from the google storage. 
+  # We add more configuration on osb.conf file template before adding it on fragments . 
+  if [ ${NGINX_OSB_HIDE_GOOGLE_HEADERS} == 1 ]; then
```
Comment (paolomainardi) [inline docker-entrypoint.sh]: Default this variable to 1 and that to CHANGELOG

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,3 +10,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - New `NGINX_HIDE_DRUPAL_HEADERS` environment variable to hide the drupal information from the response headers (default: the headers are visible)
 - New `NGINX_HIDE_SENSITIVE_HEADERS` environment variable to hide all the sensitive information from the response headers (default: the headers will be removed)
+
+## 2022-02-03
+
+### Added
+
+- New `NGINX_OSB_HIDE_GOOGLE_HEADERS` environment variable to hide the google response headers coming from the google object storage bucket (default: the headers are hidden)
```
Comment (paolomainardi) [inline CHANGELOG.md]: Just a bout the naming convention, I would prefer to be more specific here:

HIDE_GOOGLE_GCS_HEADERS

just to better understand what headers of what service we are hiding

**Comment by paolomainardi** [comment]
Still a draft, is it ready for review ?

**Pull request merged**